src/dvc_objects/transfer.py

```diff
@@ -0,0 +1,39 @@
+from typing import Set
+
+from .db import ObjectDB
+
+
+def transfer(
```

Comment:
I'm thinking about whether we even need this function. Based on how we calculate status on `dvc-data` side, the logic here will not be reusable there. Although, this will be useful for the naive transfer of objects.